product_warranty/wizard/choose_warranty.py

```diff
@@ -0,0 +1,56 @@
+from odoo import api, Command, fields, models
+
+
+class ChooseWarranty(models.TransientModel):
+    _name = 'choose.warranty'
+    _description = "Add warranty"
+
+    sale_order = fields.Many2one(
+        'sale.order',
+        string="Order",
+        required=True,
+    )
+    warranty_ids = fields.One2many(
+        'warranty.wizard.line',
+        'wizard_id',
+        string="Warranty lines",
+    )
+
+    @api.model
+    def default_get(self, fields_list):
+        res = super().default_get(fields_list)
+        sale_order = self.env['sale.order'].browse(self.env.context.get('active_id'))
+        res.update({
+            'sale_order': sale_order.id,
+            'warranty_ids': [
+                Command.create(
+                    {
+                        "sale_order_line": line.id,
+                        "warranty_id": False,
+                    })for line in sale_order.order_line.filtered(
+                        lambda line: line.product_template_id
+                        and line.product_template_id.is_warranty
+                    )
+            ]
+        })
+        return res
+
+    def action_add_warranty(self):
+        warranty_line_vals = [
+            {
+                'order_id': self.sale_order.id,
+                'product_id': self.env.ref(
+                    'product_warranty.extended_warranty_product'
+                ).id,
```

Comment:
better declare a variable
```py
product_warranty = self.env.ref('product_warranty.extended_warranty_product', raise_if_not_found=False)
```
```suggestion
                'product_id': product and product.id,
```
note: User can delete the product and using `raise_if_not_found` will return a False

In case if you want to restore the product you can use `_update_xmlids`